packages/plan-form-core/src/formHelpers/helpers.ts

```diff
@@ -172,7 +172,7 @@ export function extractActivityForForm(
       activityObj.goal.target &&
       activityObj.goal.target[0].due &&
       activityObj.goal.target[0].due !== ''
```

Comment:
Not critical but could use optional chaining and avoid the other checks. activityObj.goal?.target[0]?.due !== ''